src/realm/dictionary.cpp

```diff
@@ -668,12 +668,12 @@ auto Dictionary::erase(Iterator it) -> Iterator
     return {this, pos};
 }
 
-void Dictionary::nullify(Mixed key)
+void Dictionary::nullify(size_t ndx)
 {
     REALM_ASSERT(m_dictionary_top);
-    auto ndx = do_find_key(key);
     REALM_ASSERT(ndx != realm::npos);
 
+    auto key = do_get_key(ndx);
```

Comment:
Minor stuff, we don't need to fetch the key if the replication is not present.

Reply:
:+1: